pandas/core/frame.py

```diff
@@ -6821,6 +6821,50 @@ def corr(self, method='pearson', min_periods=1):
 
         return self._constructor(correl, index=idx, columns=cols)
 
+    def pearson_with(self, other, min_periods=1):
```

Comment:
we won't accept another top level method, but would accept a cythonized dispatch, add a method keyword similar to https://github.com/pandas-dev/pandas/pull/22375